modules/sfp_keybase.py

```diff
@@ -0,0 +1,244 @@
+          headers=headers,
+          timeout=15,
+          useragent=self.opts['_useragent']
+        )
+
+        # In this case, it will always be 200 if keybase is queried 
+        # The actual response codes are stored in status tag of the response
+        if not res['code'] == '200':
+            return None
+        
+        # Replacing null with "None"
+        content = json.loads(str(res['content']).replace(":null", ":\"None\""))
+
+        status = content.get('status')
+        if status is None:
+            return None
+
+        code = status.get('code')
+        if code is None:
+            return None
+        
+        try:
+            if not int(code) == 0:
+                return None
+        except:
+            self.sf.error("Invalid code returned as response", False)
+            return None
+
+        return content
+
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+
+        self.results[eventData] = True
+
+        userName = eventData
+
+        # Extract username if a Keybase link is received 
+        if eventName == "LINKED_URL_EXTERNAL":
+            linkRegex = "keybase.io\/[A-Za-z0-9-_.]+"  
+            link = re.findall(linkRegex, eventData)
+
+            if len(link) == 0:
+                self.sf.debug("Not a keybase link")
+                return None
+
+            userName = link.split("/")[1]
+
+            if userName in self.results:
+                return None
+            self.results[userName] = True
+
+        content = self.queryUsername(userName)
+
+        if content is None:
+            self.sf.debug("No data found for username")
+            return None
+        
+        evt = SpiderFootEvent("RAW_RIR_DATA", str(content), self.__name__, event)
+        self.notifyListeners(evt) 
+
+        if eventName == "LINKED_URL_EXTERNAL":
+            evt = SpiderFootEvent("USERNAME", str(userName), self.__name__, event)
+            self.notifyListeners(evt)    
+    
+        
+        # Replacing string values that are not enclosed within double quotes
+        # Also replacing values like True and False to their corresponding numeric values
+        # If the above steps aren't performed, json.loads() fails
+        
+        # Checking with == "None" in addition to is None, because null values are replaced with "None"
+
+        # Contains all data about the target username
+        try:
+            them = json.loads(str(content.get('them')[0]).replace("'", "\"").replace("True", "1").replace("False", "0"))
```

Comment:
I suspect the `str()` and subsequent `replace()`s are unnecessary since `content.get('them')[0]` is going to be valid JSON (if not it should raise an exception anyway).